Common/Orders/Slippage/MarketImpactSlippageModel.cs

```diff
@@ -0,0 +1,221 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using QuantConnect.Interfaces;
+using QuantConnect.Data.Consolidators;
+using QuantConnect.Data.Market;
+using QuantConnect.Securities;
+using QuantConnect.Data;
+using QuantConnect.Indicators;
+using MathNet.Numerics.Statistics;
+
+namespace QuantConnect.Orders.Slippage
+{
+    /// <summary>
+    /// Slippage model that mimic the effect brought by market impact,
+    /// i.e. consume the volume listed in the order book
+    /// </summary>
+    /// <remark>Almgren, R., Thum, C., Hauptmann, E., & Li, H. (2005). 
```

Comment:
Let's add the link to the paper:
https://www.ram-ai.com/sites/default/files/2022-06/costestim.pdf
for future reference.